core/string/unpack/u_spec.rb

```diff
@@ -51,6 +51,21 @@
     ].should be_computed_by(:unpack)
   end
 
+  it "returns an empty string if the encoded length is too low" do
+    " 86)C".unpack("u").should == [""]
+    "\x1F86)C".unpack("u").should == [""]
+    "\x0186)C".unpack("u").should == [""]
+    "\x0086)C".unpack("u").should == [""]
+  end
+
+  it "returns a max line of 45 bytes if the encoded length is too high" do
+    encoded_line = '86)C' * 15 # 60 bytes maximum
+    decoded_line = 'abc' * 15 # 45 bytes maximum
+    ("N" + encoded_line).unpack("u").should == [decoded_line]
+    ("Z" + encoded_line).unpack("u").should == [decoded_line]
+    ("_" + encoded_line).unpack("u").should == [decoded_line]
+  end
```

Comment:
TBH neither the description nor checks of the second test case help me to understand the functionality nuance. Could you please elaborate a bit more on it? (What do "too high" and "max line" mean for instance?)

Reply:
Ahh, sure! In learning about [uuencoding](https://en.wikipedia.org/wiki/Uuencoding) for the first time, I learned that the first character encodes the length of the line, which affects the result. There is a valid range for this length: it must be between 3 and 45. 

The length is encoded as the byte count + 32, which means the first character will be something between '#' (3 bytes) and 'M' (45 bytes).

Anything less than '#' or greater than 'M' is invalid, as far as I can tell.

Reply:
Uggh! There is a mistake here... I didn't actually test that the line was truncated, so this test is not testing what I thought it was. I'll come back to this later!

Reply:
Thank you for explanation :bow: